x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1168,13 +1170,13 @@ private void findTokenFromRefreshToken(
                 onFailure.accept(ex);
             }
         };
-        final SecurityIndexManager frozenTokensIndex = tokensIndexManager.freeze();
+        final SecurityIndexManager frozenTokensIndex = tokensIndexManager.defensiveCopy();
         if (frozenTokensIndex.indexExists() == false) {
             logger.warn("index [{}] does not exist so we can't find token from refresh token", frozenTokensIndex.aliasName());
-            listener.onFailure(frozenTokensIndex.getUnavailableReason());
-        } else if (frozenTokensIndex.isAvailable() == false) {
+            listener.onFailure(new IndexNotFoundException(frozenTokensIndex.aliasName()));
```

Comment:
this change emulates what getUnavailableReason() would have resulted in.... i didn't want to accept `null` in the method param for this one instance.